api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -411,6 +411,10 @@ message Bootstrap {
 
   // Optional gRPC async manager config.
   GrpcAsyncClientManagerConfig grpc_async_client_manager_config = 40;
+
+  // Optional configuration for memory allocation manager.
+  // Memory releasing is only supported for `tcmalloc allocator <https://github.com/google/tcmalloc>`_.
+  MemoryAllocatorManager memory_allocator_manager = 41;
```

Comment:
Do we think this should be configured on by default and included by all users of Envoy, or should we use the boostrap extension mechanism to make it opt in?  I know from an e-m / leanness standpoint I'd rather have it be a plug-in but I've got some bias here :-)

Reply:
There is a check in `MemoryAllocatorManager` that does not enable memory releasing if default value of `bytes_to_release_` is used. It will be noop in that case. Realistically we will only support one allocator type in the long run, so that undermines the purpose of having it as extension. On the other hand it would make this feature more explicit for users to configure.

Reply:
yeah in general I prefer lose coupling (using Any extensions) to tight coupling (which pulls in all deps) as it fits with our long term desire to more xDS in a more flexible direction.  That said this is small and gRPC doesn't use bootstrap so it's only a mild preference unless  @htuch or @markdroth think it's worth doing

Reply:
Yeah, no strong opinion here since bootstrap will always be dataplane specific.